index.js

```diff
@@ -17,7 +19,13 @@ module.exports = function (str, filepath) {
 
 	return pify(mkdirp, Promise)(path.dirname(fullpath))
 		.then(function () {
-			return pify(fs.writeFile, Promise)(fullpath, str);
+			return isStream(str) ?
+				new Promise((resolve, reject) => {
+					str.pipe(fs.createWriteStream(fullpath))
```

Comment:
> The .on('error', reject) below is handling errors on the writeable stream.

I see, I'm always confused by streams. But then I guess we need an error handler on the user specified stream (`str`) then. ?

> As for the code style. Like this?

👍